Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM rust:1.77.0 as builder
+LABEL Description="OpenLan CGW (Build) environment"
+
+WORKDIR /usr/src/openlan-cgw
+
+RUN rustup target add x86_64-unknown-linux-gnu
+
+RUN apt-get update -q -y  && apt-get install -q -y \
+	pkg-config \
+	build-essential \
+	cmake \
+	protobuf-compiler \
+	libssl-dev
+
+CMD ["make", "-C", "/usr/src/openlan-cgw", "cgw-app"]
```

Comment:
Could this be setup as a `RUN`?
`RUN make -C /usr/src/openlan-cgw cgw-app`

Reply:
# Build the application.
# Leverage a cache mount to /usr/local/cargo/registry/
# for downloaded dependencies, a cache mount to /usr/local/cargo/git/db
# for git repository dependencies, and a cache mount to /app/target/ for
# compiled dependencies which will speed up subsequent builds.
# Leverage a bind mount to the src directory to avoid having to copy the
# source code into the container. Once built, copy the executable to an
# output directory before the cache mounted /app/target is unmounted.
RUN --mount=type=bind,source=src,target=src \
    --mount=type=bind,source=Cargo.toml,target=Cargo.toml \
    --mount=type=bind,source=Cargo.lock,target=Cargo.lock \
    --mount=type=cache,target=/app/target/ \
    --mount=type=cache,target=/usr/local/cargo/git/db \
    --mount=type=cache,target=/usr/local/cargo/registry/ \
cargo build --locked --release && \
cp ./target/release/$APP_NAME /bin/server


also i do note you have a Makefile for all clean run-cgw cgw-img stop-cgw

was there a specific reason for this?